### PR TITLE
Don't start the sync when not needed

### PR DESCRIPTION
**Pull request by gumb0**
In the private network with two nodes, when they connected, the sync was started on both nodes. The node with higher difficulty shouldn't start the sync.

### CODE REVIEW

File: libweb3jsonrpc/IpcServerBase.cpp [hunk 1]
```diff
@@ -36,6 +36,7 @@ int const c_bufferSize = 1024;
 template <class S> IpcServerBase<S>::IpcServerBase(string const& _path):
     m_path(_path)
 {
+    clog(VerbosityInfo, "rpc") << "JSONRPC socket path: " << _path;
```
Comment (chfast) [inline libweb3jsonrpc/IpcServerBase.cpp]: `JSON-RPC`.

File: libethereum/BlockChainSync.cpp [hunk 2]
```diff
@@ -689,6 +690,7 @@ void BlockChainSync::collectBlocks()
         case ImportResult::UnknownParent:
             if (headers.first + i > m_lastImportedBlock)
             {
+                LOG(m_logger) << "Already known or future time unknown or unknown parent, block #" << headers.first + i << ". Resetting sync.";
```
Comment (chfast) [inline libethereum/BlockChainSync.cpp]: clang-format

File: libethereum/BlockChainSync.cpp [hunk 3]
```diff
@@ -676,6 +676,7 @@ void BlockChainSync::collectBlocks()
             break;
         case ImportResult::Malformed:
         case ImportResult::BadChain:
+            LOG(m_logger) << "Malformed or bad chain, block #" << headers.first + i << ". Restarting sync.";
```
Comment (chfast) [inline libethereum/BlockChainSync.cpp]: Can you separate the logs? One per "Malformed", one per "BadChain".

**Comment by gumb0** [comment]
@chfast fixed

**Pull request merged**